spopt/locate/frlm.py

```diff
@@ -0,0 +1,2089 @@
+        if facility not in combination:
+            return 0
+        
+        if facility == origin or facility == destination:
+            return 1
+        
+        return 2
+    
+    def check_path_refueling_feasibility(self, path: List, facilities: List) -> bool:
+        """
+        Check if a path can be traversed with given facilities. True if path is feasible, False otherwise. 
+        """
+        if len(path) < 2:
+            return False
+
+        total_length = sum(self.network[path[i]][path[i+1]]['length'] 
+                          for i in range(len(path) - 1))
+
+        if total_length * 2 <= self.vehicle_range:
+            return any(facility in path for facility in facilities)
+
+        remaining_range = self.vehicle_range / 2
+        if path[0] in facilities:
+            remaining_range = self.vehicle_range
+        
+        for i in range(1, len(path)):
+            segment_length = self.network[path[i-1]][path[i]]['length']
+            remaining_range -= segment_length
+            
+            if remaining_range < 0:
+                return False
+            
+            if path[i] in facilities:
+                remaining_range = self.vehicle_range
+
+        remaining_range = self.vehicle_range / 2
+        if path[-1] in facilities:
+            remaining_range = self.vehicle_range
+        
+        for i in range(len(path) - 1, 0, -1):
+            segment_length = self.network[path[i-1]][path[i]]['length']
+            remaining_range -= segment_length
+            
+            if remaining_range < 0:
+                return False
+            
+            if path[i-1] in facilities:
+                remaining_range = self.vehicle_range
+        
+        return True
+    
+    def generate_path_refueling_combinations(self) -> Dict:
+        """
+        Generate dictionary mapping OD pairs to valid facility combinations
+        """
+        if not self.facility_combinations:
+            self.generate_facility_combinations()
+        
+        path_refueling_combinations = {}
+        
+        for od_pair, path in tqdm(self.flow_paths.items(), desc="Generating combinations"):
+            valid_combinations = []
+            
+            for combo in self.facility_combinations:
+                if self.check_path_refueling_feasibility(path, combo):
+                    valid_combinations.append(combo)
+            
+            path_refueling_combinations[od_pair] = valid_combinations
+
+            self.compute_refueling_frequency(od_pair[0], od_pair[1])
+        
+        self.path_refueling_combinations = path_refueling_combinations
+        
+        total_valid = sum(len(combinations) for combinations in path_refueling_combinations.values())
+        print(f"Generated {total_valid} valid combinations across {len(path_refueling_combinations)} OD pairs")
+        
+        return path_refueling_combinations
+    
+    def calculate_node_weights(self, method: str ='double') -> np.ndarray:
```

Comment:
Is `double/single` what's used in the literature? I think this should be implemented as `include_destination=False`.